### PR TITLE
update and modify based on physical test

### PR DESCRIPTION
**Pull request by ipa-rwu**

### CODE REVIEW

File: mobika2_tests/launch/test4_static_obstacle.launch [hunk 1]
```diff
@@ -99,10 +107,33 @@
...
-    <node pkg="araig_calculators" type="rosbagger_node" name="bag_video_node">
+    <!-- logging -->
+    <!-- rosbagger_node -->
+    <node pkg="araig_calculators" type="rosbagger_node" name="bag_video_node" output="log">
+        <remap from="/start" to="/signal/ui/start_test"/>
+        <remap from="/stop" to="/signal/runner/test_completed"/>
+    </node>
+
+    <!-- folder_manager_node -->
+    <node pkg="araig_calculators" type="folderlogger_node" name="folder_manager_node" output="log">
```
Comment (ipa-kut) [inline mobika2_tests/launch/test4_static_obstacle.launch:118]: This needs to be updated to reflect the merged folder_rosbagger node.

Comment (ipa-kut) [reply mobika2_tests/launch/test4_static_obstacle.launch:118]: In all launch files.

**Review by ipa-kut — CHANGES_REQUESTED** [review]
Please rebase the PR to update the commit messages to be a bit more verbose, and squash them as well if necessray.

**Comment by ipa-rwu** [comment]
* remove disable mavel

File: mobika2_tests/config/test1_braking.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  robot_reached_max_vel_node: # node name 
+    param: 0.25
+    tolerance: 0.06
+  
+  robot_has_stopped_node: # node name 
+    param: 0.0
+    tolerance: 0.001
+
+  braking_time_node:
+    if_log: "true"
```
Comment (ipa-kut) [inline mobika2_tests/config/test1_braking.yaml:24]: Check how to read and parse this param properly as a bool

File: mobika2_tests/launch/test1_braking.launch [hunk 3]
```diff
@@ -0,0 +1,105 @@
+<launch>
+    <arg name="disable_marvel" default="false"/>
+
+    <rosparam command="load" file="$(find mobika2_tests)/config/test1_braking.yaml" />
+
+<!-- sensors -->
+    <include file="$(find mobika2_tests)/launch/includes/common.launch">
+        <args name="src1" value="$(arg cam1)" />
```
Comment (ipa-kut) [inline mobika2_tests/launch/test1_braking.launch:8]: Add a declaration for these args in the beginning with default values.

File: mobika2_tests/launch/test1_braking.launch [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    <!-- ONLY for TEST use /odom as the location of robot -->
+    <group if="$(arg disable_marvel)"> 
+        <!-- use /odom as the location of robot, mobika ddoes not have amcl-->
+        <node name="focus_pose" pkg="topic_tools" type="transform" args="/odom /data/interpreter/location/robot geometry_msgs/PoseStamped 
+        'geometry_msgs.msg.PoseStamped(header=std_msgs.msg.Header(seq=m.header.seq,stamp=m.header.stamp),pose=m.pose.pose)' --import geometry_msgs std_msgs" />
+    </group>
+
+    <group unless ="$(arg disable_marvel)">
+        <!-- X Y Z Y P R transform of marvel origin frame wrt map frame. Needs to be physically measured. -->
+        <arg name="origin_pos" default="1.52 -2.37 0 0.55 0 0"/>
```
Comment (ipa-kut) [inline mobika2_tests/launch/test1_braking.launch:37]: not needed for braking test

File: mobika2_tests/launch/test1_braking.launch [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+        <!-- Publish static TF of the origin anchor to map -->
+        <node pkg="tf" type="static_transform_publisher" name="marvel_tf" args=" $(arg origin_pos)  /map /marvel_origin 1"/>
```
Comment (ipa-kut) [inline mobika2_tests/launch/test1_braking.launch:40]: not needed

File: mobika2_tests/launch/test1_braking.launch [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+        <node pkg="marvelmind_nav" type="hedge_rcv_bin" name="hedge_rcv_bin" />
```
Comment (ipa-kut) [inline mobika2_tests/launch/test1_braking.launch:42]: already started in common.launch

File: mobika2_tests/launch/test1_braking.launch [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+    </node>
+
+    <!-- logging -->
+    <!-- rosbagger_node -->
+    <node pkg="araig_calculators" type="rosbagger_node" name="bag_video_node" output="log">
+        <remap from="/start" to="/signal/ui/start_test"/>
+        <remap from="/stop" to="/signal/runner/test_completed"/>
+    </node>
+
+    <!-- folder_manager_node -->
```
Comment (ipa-kut) [inline mobika2_tests/launch/test1_braking.launch:83]: merge with rosbagger

File: mobika2_tests/launch/test1_emergency.launch [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+    </group>
+
+    <!-- pub emergency signal -->
+    <node pkg="rostopic" type="rostopic" name="emergency_node" args="pub /emergency_stop_state cob_msgs/EmergencyStopState
+        '{emergency_button_stop: false,
+        scanner_stop: false,
+        emergency_state: 0}'" output="screen"/> 
+
+    <!-- delay 1s transfer emergency to signal-->
+    <include file="$(find mobika2_tests)/launch/includes/timed_roslaunch.launch">
```
Comment (ipa-kut) [inline mobika2_tests/launch/test1_emergency.launch:55]: Can replace this entire feature with a topic_transformer to convert to signal

File: mobika2_tests/launch/test4_no_goal.launch [hunk 9]
```diff
@@ -0,0 +1,64 @@
+<launch>
```
Comment (ipa-kut) [inline mobika2_tests/launch/test4_no_goal.launch:1]: This launch file can be removed completely

File: mobika2_tests/launch/test4_static_obstacle.launch [hunk 10]
```diff
@@ -0,0 +1,139 @@
+<launch>
```
Comment (ipa-kut) [inline mobika2_tests/launch/test4_static_obstacle.launch:1]: Test 4 is a test of the navigation stack.    
Purpose:
* Test pose repeatability
* Test time to goal
Static obstacle not relevant

File: mobika2_tests/config/test4_static_obstacle.yaml [hunk 11]
```diff
@@ -0,0 +1,112 @@
+runner:
+  timeout_check_goal: 2
+  timeout_after_complete: 3
+  
+interpreters: 
+  goal_interpreter_node: # node name 
+    goal:
```
Comment (ipa-kut) [inline mobika2_tests/config/test4_static_obstacle.yaml:7]: Read these params from interpreter namespace directly (bypass node name)

File: mobika2_tests/launch/test4_static_obstacle.launch [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+    <node pkg="araig_interpreters" type="goal_interpreter" name="goal_interpreter_node" output="screen">
+        <remap from="/goal" to="/data/interpreter/goal"/>
+        <remap from="/start" to="/signal/runner/start_robot"/>
+    </node>
+
+    <!-- goal frame -->
+    <node pkg="araig_interpreters" type="pose_tf_broadcaster" name="goal_tf2_broadcaster_node" output="screen" />
+
+    <!-- static obstacle frame -->
+    <node pkg="araig_interpreters" type="pose_tf_broadcaster" name="obstacle_tf2_broadcaster_node" output="screen" />
```
Comment (ipa-kut) [inline mobika2_tests/launch/test4_static_obstacle.launch:28]: Not required.

File: mobika2_tests/launch/test4_static_obstacle.launch (hunk 10 above)
Comment (ipa-kut) [reply mobika2_tests/launch/test4_static_obstacle.launch:1]: Remove all static obstacle parts

**Review by ipa-kut — CHANGES_REQUESTED** [review]

**Pull request closed without merging**